File: airchainpay-wallet/src/types/token.ts
import { ImageSourcePropType } from 'react-native';

export interface TokenInfo {
  symbol: string;
  name: string;
  address: string;
  decimals: number;
  chainId: string;
  chainName: string;
  logoUri?: ImageSourcePropType | string;
  isNative: boolean;
  isStablecoin?: boolean;
  contractAddress?: string;
  balance?: string;
  usdValue?: number;
}

File: airchainpay-wallet-core/src/domain/entities/token.rs
//! Token entity for the wallet core

use serde::{Serialize, Deserialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct TokenInfo {
    pub symbol: String,
    pub name: String,
    pub decimals: u8,
    pub address: String,
    pub chain_id: u64,
    pub is_native: bool,
    pub is_stablecoin: bool,
} 